addons/udes_stock_cron/models/stock_picking_type.py

```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
+from odoo import fields, models
+
+
+class StockPickingType(models.Model):
+    _inherit = "stock.picking.type"
+
+    u_num_reservable_pickings = fields.Integer(
+        string="Number of Pickings to Reserve",
+        default=0,
+        help="The number of pickings in the queue to reserve stock for. "
+        "If batch reservation is enabled, entire picking batches are "
+        "reserved in the order of their earliest picking in the queue "
+        "until at least this number of pickings are reserved. "
+        "0 indicates no pickings should be reserved. "
+        "-1 indicates all pickings should be reserved.",
```

Comment:
Might be worth considering if there is a nicer way to do this. Looking at the config screen it isn't clear that `-1` means reserve all confirming pickings.

Personally I think it would look nicer to have a boolean field for saying whether a picking type gets reserved via a cron job, then another field for saying how many, but adding an extra field isn't exactly nice...

Can ignore this comment for now. :)

Reply:
Yep we can tackle this later